update-bcd.ts

```diff
@@ -461,6 +461,18 @@ export const update = (
         simpleStatement.version_added !== inferredStatement.version_added
       ) {
         simpleStatement.version_added = inferredStatement.version_added;
+
+        // When a "mirrored" statement will be replaced with a statement
```

Comment:
I prefer that more functional style since (as is often the case with FP) it's easier to read and probably more correct. It handles cases that I hadn't considered, like when the upstream compat data uses the `impl_url` property.

I'm inclined to update the implementation (and extend the test), but there's another edge case I'd like to verify: if there are other statements besides the one which uses `partial_implementation`:

    chrome: [
      {version_added: "85", partial_implementation: true, notes: ""},
      {version_added: "84", flags: [{}]}
    ],
    chrome_android: "mirror"

Given a negative test result from Chrome for Android version 85, should the result be

```diff
 chrome: [
   {version_added: "85", partial_implementation: true, notes: ""},
   {version_added: "84", flags: [{}]}
 ],
-chrome_android: "mirror"
+chrome_android: [
+  {version_added: false},
+  {version_added: "84", flags: [{}]}
+]
```

or

```diff
 chrome: [
   {version_added: "85", partial_implementation: true, notes: ""},
   {version_added: "84", flags: [{}]}
 ],
-chrome_android: "mirror"
+chrome_android: {version_added: false}
```

?

Reply:
Good question! I think the correct data can't be reliably inferred automatically here, but that the second is preferable since it doesn't contain any false claims but the first one might. To elaborate, this browser has done something to disable a feature that's enabled by default in the upstream. That something was probably disabling the flag, but it could also be removing code for binary size so that it's not actually possible to enable.

The output of this script will still be reviewed by a human, so there is a chance to tweak the data if someone has more information than we can infer here.

Reply:
Thanks for the clarification!